File: 프로그래머스/python/LV3/야근 지수.py
import heapq


def solution(n, works):
    answer = 0
    q = []
    if n >= sum(works):
        return 0
    for work in works:
        heapq.heappush(q, -work)
    for _ in range(n):
        maxV = heapq.heappop(q)
        heapq.heappush(q, maxV + 1)

    for i in q:
        answer += i**2
    return answer

File: 프로그래머스/javascript/LV3/야근지수.js
const solution = (n, works) => {
  works.sort((a, b) => b - a);
  let idx = 0;
  while (n > 0) {
    if (works[idx] <= works[0]) idx = 0;
    if (works[idx] === 0 || idx >= works.length) break;
    works[idx]--;
    n--;
    idx++;
    if (idx >= works.length) idx = 0;
  }
  console.log(works);
  return works.reduce((a, c) => a + c * c, 0);
};
